plugin-energy-attribution/src/lib.rs

```
@@ -67,22 +72,27 @@ impl AlumetPlugin for EnergyAttributionPlugin {
     }
 
     fn pre_pipeline_start(&mut self, alumet: &mut AlumetPreStart) -> anyhow::Result<()> {
-        /// Finds the RawMetricId with the name of the metric.
-        /// Will only run once, just before the pipeline starts.
-        fn find_metric_by_name(alumet: &mut AlumetPreStart, name: &str) -> anyhow::Result<RawMetricId> {
-            let (id, _metric) = alumet
-                .metrics()
-                .into_iter()
-                .find(|m| m.1.name == name)
-                .unwrap_or_else(|| panic!("Cannot find metric {name}, are the 'rapl' and 'k8s' plugins loaded?"));
-            Ok(id.to_owned())
-        }
-
+        // Finds the RawMetricId with the name of the metric.
+        // Will only run once, just before the pipeline starts.
         // Lock the metrics mutex to apply its modifications.
         let mut metrics = self.metrics.lock().unwrap();
-
-        metrics.rapl_consumed_energy = Some(find_metric_by_name(alumet, "rapl_consumed_energy")?);
-        metrics.cpu_usage_per_pod = Some(find_metric_by_name(alumet, "cgroup_cpu_usage_user")?);
+        let tmp_metrics_consumed_energy = alumet.metrics();
+        let tmp_metrics_hardware_usage = alumet.metrics();
+
+        metrics.consumed_energy = Some(
+            tmp_metrics_consumed_energy
+                .by_name(&self.config.consumed_energy)
+                .take()
+                .unwrap()
+                .0,
```

That's a bit convoluted IMO. Also, you should return an error message if the metric is not found (use `context` or `with_context`)

---

same here, these lines are irrelevant